test/spec/Feature/Query/DeleteSpec.hs

```diff
@@ -115,3 +115,81 @@ spec =
             { matchStatus = 204
             , matchHeaders = [matchHeaderAbsent hContentType]
             }
+
+    context "limited delete" $ do
+      it "works with the limit and offset query params" $
+        request methodDelete "/projects?limit=1&offset=4&select=id,name"
+            [("Prefer", "return=representation")]
+            mempty
+          `shouldRespondWith`
+            [json|[
+              {
+                  "id": 5,
+                  "name": "Orphan"
+              }]|]
+            { matchStatus = 200 }
```

Comment:
I haven't had the time to look at the code, yet. But those test cases seem like they don't really test what we need to test.

>  - #2156, `limit/offset` now limits the affected rows on UPDATE/DELETE  - @steve-chavez
>   + Previously, `limit/offset` only limited the returned rows but not the actual updated rows

The returned rows are not really changed. They were the same before and after your change.

We'll need to test that the actual `UPDATE` or `DELETE` really affected only those rows, too. I guess we'll need some requests with `Prefer: tx=commit`, then check for which rows exist in the database, then rollback the changes manually. Because it's actual mutations to the database, it would be better to create a new table for those tests then, to not conflict with other test cases running on the same fixtures.

Reply:
True. Fair enough, I'll update the tests.

Reply:
Ok done. I must say, doing tests for real mutations is too tedious/painful.. we need utilitarian functions for this. Not willing to do that here though, could be added later.